File: Navigation/AppPageDescriptor.vb
Imports WpfApp1.Infrastructure

Namespace Navigation
    Public NotInheritable Class AppPageDescriptor
        Inherits BindableBase

        Private _displayTitle As String
        Private _displaySummary As String
        Private _displayMenuTitle As String
        Private _displayMenuMeta As String
        Private _menuIndexText As String
        Private _isCurrentPage As Boolean

        Public Sub New(pageKey As AppPageKey,
                       titleResourceKey As String,
                       fallbackTitle As String,
                       summaryResourceKey As String,
                       fallbackSummary As String,
                       menuMetaResourceKey As String,
                       fallbackMenuMeta As String,
                       pageViewModel As Object)
            Me.PageKey = pageKey
            Me.TitleResourceKey = titleResourceKey
            Me.FallbackTitle = fallbackTitle
            Me.SummaryResourceKey = summaryResourceKey
            Me.FallbackSummary = fallbackSummary
            Me.MenuMetaResourceKey = menuMetaResourceKey
            Me.FallbackMenuMeta = fallbackMenuMeta
            Me.PageViewModel = pageViewModel

            _displayTitle = fallbackTitle
            _displaySummary = fallbackSummary
            _displayMenuTitle = fallbackTitle
            _displayMenuMeta = fallbackMenuMeta
            _menuIndexText = String.Empty
        End Sub

        Public ReadOnly Property PageKey As AppPageKey

        Public ReadOnly Property TitleResourceKey As String

        Public ReadOnly Property FallbackTitle As String

        Public ReadOnly Property SummaryResourceKey As String

        Public ReadOnly Property FallbackSummary As String

        Public ReadOnly Property MenuMetaResourceKey As String

        Public ReadOnly Property FallbackMenuMeta As String

        Public ReadOnly Property PageViewModel As Object

        Public Property DisplayTitle As String
            Get
                Return _displayTitle
            End Get
            Private Set(value As String)
                SetProperty(_displayTitle, value)
            End Set
        End Property

        Public Property DisplaySummary As String
            Get
                Return _displaySummary
            End Get
            Private Set(value As String)
                SetProperty(_displaySummary, value)
            End Set
        End Property

        Public Property DisplayMenuTitle As String
            Get
                Return _displayMenuTitle
            End Get
            Private Set(value As String)
                SetProperty(_displayMenuTitle, value)
            End Set
        End Property

        Public Property DisplayMenuMeta As String
            Get
                Return _displayMenuMeta
            End Get
            Private Set(value As String)
                SetProperty(_displayMenuMeta, value)
            End Set
        End Property

        Public Property MenuIndexText As String
            Get
                Return _menuIndexText
            End Get
            Private Set(value As String)
                SetProperty(_menuIndexText, value)
            End Set
        End Property

        Public Property IsCurrentPage As Boolean
            Get
                Return _isCurrentPage
            End Get
            Private Set(value As Boolean)
                SetProperty(_isCurrentPage, value)
            End Set
        End Property

        Public Sub SetDisplayText(title As String, summary As String)
            DisplayTitle = title
            DisplaySummary = summary
        End Sub

        Public Sub SetMenuDisplayText(menuTitle As String, menuMeta As String, menuIndexText As String)
            DisplayMenuTitle = menuTitle
            DisplayMenuMeta = menuMeta
            MenuIndexText = menuIndexText
        End Sub

        Public Sub SetCurrentPageState(isCurrentPage As Boolean)
            IsCurrentPage = isCurrentPage
        End Sub
    End Class
End Namespace

File: src/ClickSyncMouseTester/Navigation/AppPageDescriptor.cs
using ClickSyncMouseTester.Infrastructure;

namespace ClickSyncMouseTester.Navigation;

public sealed class AppPageDescriptor : BindableBase
{
    private string _displayTitle;

    private string _displaySummary;

    private string _displayMenuTitle;

    private string _displayMenuMeta;

    private string _menuIndexText;

    private bool _isCurrentPage;

    public AppPageKey PageKey { get; }

    public string TitleResourceKey { get; }

    public string FallbackTitle { get; }

    public string SummaryResourceKey { get; }

    public string FallbackSummary { get; }

    public string MenuMetaResourceKey { get; }

    public string FallbackMenuMeta { get; }

    public object PageViewModel { get; }

    public string DisplayTitle
    {
        get
        {
            return _displayTitle;
        }
        private set
        {
            SetProperty(ref _displayTitle, value, "DisplayTitle");
        }
    }

    public string DisplaySummary
    {
        get
        {
            return _displaySummary;
        }
        private set
        {
            SetProperty(ref _displaySummary, value, "DisplaySummary");
        }
    }

    public string DisplayMenuTitle
    {
        get
        {
            return _displayMenuTitle;
        }
        private set
        {
            SetProperty(ref _displayMenuTitle, value, "DisplayMenuTitle");
        }
    }

    public string DisplayMenuMeta
    {
        get
        {
            return _displayMenuMeta;
        }
        private set
        {
            SetProperty(ref _displayMenuMeta, value, "DisplayMenuMeta");
        }
    }

    public string MenuIndexText
    {
        get
        {
            return _menuIndexText;
        }
        private set
        {
            SetProperty(ref _menuIndexText, value, "MenuIndexText");
        }
    }

    public bool IsCurrentPage
    {
        get
        {
            return _isCurrentPage;
        }
        private set
        {
            SetProperty(ref _isCurrentPage, value, "IsCurrentPage");
        }
    }

    public AppPageDescriptor(AppPageKey pageKey, string titleResourceKey, string fallbackTitle, string summaryResourceKey, string fallbackSummary, string menuMetaResourceKey, string fallbackMenuMeta, object pageViewModel)
    {
        PageKey = pageKey;
        TitleResourceKey = titleResourceKey;
        FallbackTitle = fallbackTitle;
        SummaryResourceKey = summaryResourceKey;
        FallbackSummary = fallbackSummary;
        MenuMetaResourceKey = menuMetaResourceKey;
        FallbackMenuMeta = fallbackMenuMeta;
        PageViewModel = pageViewModel;
        _displayTitle = fallbackTitle;
        _displaySummary = fallbackSummary;
        _displayMenuTitle = fallbackTitle;
        _displayMenuMeta = fallbackMenuMeta;
        _menuIndexText = string.Empty;
    }

    public void SetDisplayText(string title, string summary)
    {
        DisplayTitle = title;
        DisplaySummary = summary;
    }

    public void SetMenuDisplayText(string menuTitle, string menuMeta, string menuIndexText)
    {
        DisplayMenuTitle = menuTitle;
        DisplayMenuMeta = menuMeta;
        MenuIndexText = menuIndexText;
    }

    public void SetCurrentPageState(bool isCurrentPage)
    {
        IsCurrentPage = isCurrentPage;
    }
}






